app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -50,15 +56,30 @@ define(function (require, exports, module) {
           return new ProfileImage();
         })
         .then(function (profileImage) {
+          if (showSpinner) {
+            // Wait for an upper limit of MAX_SPINNER_COMPLETE_TIME before
+            // showing the profile image.  May only be necessary to satisfy
+            // tests.
+            return p.allSettled([
```

Comment:
Is `p.allSettled` needed here? Could you just return `self._completeLoadingSpinner` since it returns a promise itself? In any case, `allSettled` seems like the wrong method, only one promise is being checked.